app/src/full/kotlin/io/homeassistant/companion/android/data/wear/WearDataListener.kt

```diff
@@ -0,0 +1,36 @@
+package io.homeassistant.companion.android.data.wear
+
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.wearable.WearableListenerService
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.tasks.asTask
+import okhttp3.Dns
+
+@AndroidEntryPoint
+class WearDataListener : WearableListenerService() {
```

Comment:
How do you envision this class to evolve? The name seems quite generic. Could you add a description within the code of the scope of the listener?

Reply:
Yep, happy for guidance here. 

I'll document the current scope (mobile providing helper APIs for networking/data).

Reply:
In that case if you envision it to be for general networking/data I would suggest to follow [Google documentation ](https://developers.google.com/android/reference/com/google/android/gms/wearable/WearableListenerService)

> For best performance, limit the number of listeners and use mutually exclusive filters where there are multiple listeners. A common pattern is to use prefix filters to listen to different namespaces within an application.

And introduce a namespace in the intent-filter something like `android:pathPrefix="/network"` and update the path to look like `/network/dnsLookup`